rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -1994,4 +1912,248 @@ mod tests {
         assert_eq!(batch.schema().as_ref(), &expected_schema);
         assert_eq!(batch, expected_batch);
     }
+
+    // Property 1: for all values, if an array is entirely that value then it
+    // should become both the min and max. There are some exceptions:
+    // * NaN, null are ignored for min and max
+    // * string / binary more than 64 bytes are truncated.
+    fn assert_min_max_constant_property(
+        value: ScalarValue,
+        with_nulls: bool,
+    ) -> std::result::Result<(), TestCaseError> {
+        let array_scalar = value.to_scalar();
+        let (array, _) = array_scalar.get();
+        let array = make_array(array.to_data());
+
+        let array = if with_nulls {
+            let nulls = new_null_array(array.data_type(), 1);
+            interleave(&[&array, &nulls], &[(1, 0), (0, 0)]).unwrap()
+        } else {
+            array
+        };
+
+        let stats = collect_statistics(&[&array]);
+        prop_assert_eq!(
+            stats,
+            StatisticsRow {
+                null_count: if with_nulls { 1 } else { 0 },
+                min_value: value.clone(),
+                max_value: value,
+            },
+            "Statistics are wrong for input data: {:?}",
+            array
+        );
+        Ok(())
+    }
+
+    #[test]
+    fn test_min_max_constant_property() {
+        let values = [
+            ScalarValue::Boolean(Some(true)),
+            ScalarValue::Boolean(Some(false)),
+            ScalarValue::Int32(Some(1)),
+            ScalarValue::Float32(Some(1.0)),
+            ScalarValue::Float32(Some(std::f32::INFINITY)),
+            ScalarValue::Float32(Some(std::f32::NEG_INFINITY)),
+            ScalarValue::Utf8(Some("foo".to_string())),
+            ScalarValue::Utf8(Some("a".repeat(BINARY_PREFIX_LENGTH))),
+            ScalarValue::Binary(Some(vec![0_u8; BINARY_PREFIX_LENGTH])),
+        ];
+
+        for value in values {
+            assert_min_max_constant_property(value.clone(), false).unwrap();
+            assert_min_max_constant_property(value.clone(), true).unwrap();
+        }
+    }
+
+    proptest::proptest! {
+        #[test]
+        fn test_min_max_constant_property_timestamp(
+            timezone_index in 0..3_usize,
+            timeunit_index in 0..4_usize,
+        ) {
+            let value = Some(42);
+            // Check different timezones to validate we propagate them.
+            let timezones = [None, Some("UTC".into()), Some("America/New_York".into())];
```

Comment:
Actually for the purpose of the test timezone could be any string? This is nicer to read.

Reply:
It didn't feel relevant to test more than these three options, since we don't do any parsing of them.

Reply:
Oh yeah, just `[None, Some("UTC".into())]` would be enough imo, but this is nice to read.